src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/DebugInfoTableNode.cs

```diff
@@ -83,15 +83,14 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
 
             foreach (MethodDesc method in ((ReadyToRunTableManager)factory.MetadataManager).GetCompiledMethods())
             {
-                MethodWithGCInfo methodCodeNode = factory.MethodEntrypoint(method) as MethodWithGCInfo;
-                if (methodCodeNode == null)
+                IMethodNode methodNode = factory.MethodEntrypoint(method);
+                MethodWithGCInfo methodCodeNode = methodNode as MethodWithGCInfo;
+                if (methodCodeNode == null && methodNode is LocalMethodImport localMethodImport)
```

Comment:
Would it be worth encapsulating this logic we run through in each node that uses the metadata manager to get a list of compiled methods? We have 5 places where we go through very similar steps.

Reply:
Good point, I've also been thinking about this while fixing these multiple places. I'll look into unifying these.